components/locale_canonicalizer/src/locale_canonicalizer.rs

```diff
@@ -186,12 +186,28 @@ fn update_langid(
     entry: &LanguageIdentifier,
     langid: &mut LanguageIdentifier,
 ) -> CanonicalizationResult {
-    if langid.language.is_empty() {
+    let mut modified: bool = false;
```

Comment:
`nit (n-b)`: you don't need to specify the type, its derived from the initial value.